src/state/base_state.go

```diff
@@ -92,11 +92,11 @@ func (bs *BaseState) ApplyTransaction(
 	txIndex int,
 	blockHash common.Hash,
 	coinbase common.Address,
-	noReceipt bool) (*ethTypes.Receipt, error) {
+	noReceipt bool) (*ethTypes.Receipt, common.Address, error) {
```

Comment:
I would be tempted to update this function to receive an evmlTransaction, and not have to return the from address

Reply:
I agree. I'll make the changes.